src/ATokenNaive.sol

```diff
@@ -198,15 +203,24 @@ contract ATokenNaive is AToken {
     uint256 amount,
     uint256 index
   ) internal returns (bool) {
-    bool _returnVar = _mintScaled(caller, onBehalfOf, amount, index);
-
     // We increment by `amount` instead of any computed/rebased amounts because
     // `amount` is what actually gets transferred of the underlying asset. We
-    // need our checkpoints to still match up with the underlying asset balance.
-    _writeCheckpoint(_checkpoints[onBehalfOf], _additionFn, amount);
-    _writeCheckpoint(_totalDepositCheckpoints, _additionFn, amount);
+    // need our checkpoints to still match up with underlying asset transactions.
+    Checkpoints.History storage _depositHistory = _depositCheckpoints[onBehalfOf];
+    Checkpoints.push(_depositHistory, Checkpoints.latest(_depositHistory) + amount);
```

Comment:
Optional syntax change, since `Checkpoints` is a library, you can do `using Checkpoints for Checkpoints.History` at the top of the contract, then this line (and other similar lines) becomes:
```suggestion
    Checkpoints.push(_depositHistory, _depositHistory.latest() + amount);
```